docs/mget.1
.\" Automatically generated by Pod::Man v1.37, Pod::Parser v1.32
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "MGET 1"
.TH MGET 1 "2007-11-03" "perl v5.8.8" "User Contributed Perl Documentation"
.SH "NAME"
mget \- download videos off video hosting sites
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
mget [options...] [url]
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
Movie Get aims to be a simple console script allowing a user to download embeded movies from various video hosting services. It uses common *nix programs to download the file and convert it from flv to mpg format. The main work of mget is to get the direct video link, which it achieves with the help of the great ruby programming language.
.SH "OPTIONS"
.IX Header "OPTIONS"
\&\fB\-h\fR
\&\fB\-\-help\fR
  Display a short help text and exits
.PP
\&\fB\-n\fR
\&\fB\-\-name\fR
  Save the video with the specified name. If no name is supplied the file will be named using the scheme: MovieSiteXXXXXX where \s-1XXXXXX\s0 is a unique integer. This flag doesn't work if input is taken from a file. The file extension will be added automatically.
.PP
\&\fB\-i\fR
\&\fB\-\-input\fR
  Reads movie links from a file. Each link should be on one line by itself. With this option all movies will be downloaded into directories named after the hosting site. In example movies downloaded from Youtube will go to Youtube/YoutubeXXXXXX where \s-1XXXXXX\s0 is an unique integer.
.PP
\&\fB\-d\fR
\&\fB\-\-download\fR
  If this flag is set the script downloads videos without prompting the user.
.PP
\&\fB\-D\fR
\&\fB\-\-nodownload\fR
  If this flag is set then no files are downloaded.
  This flag also automatically sets the \fB\-C\fR and \fB\-s\fR flags.
.PP
\&\fB\-c\fR
\&\fB\-\-convert\fR
  Converts all downloaded files that are convertable without prompting the user.
  This flag also automatically sets the \fB\-d\fR flag
.PP
\&\fB\-C\fR
\&\fB\-\-noconvert\fR
  If this flag is set then no files are converted.
.PP
\&\fB\-r\fR
\&\fB\-\-remove\fR
  If this flag is set then leftover files from conversion are removed without prompting the user.
.PP
\&\fB\-R\fR
\&\fB\-\-noremove\fR
  If this flag is set then \fBno\fR leftover files are removed and the user is not prompted.
.PP
\&\fB\-s\fR
\&\fB\-\-show\fR
  If this flag is set then the direct link to download the movie is printed on the screen. You can then take this link and paste it directly into your browser to start the download.
.PP
\&\fB\-q\fR
\&\fB\-\-quiet\fR
  This flag hides output from wget (and wget only).
.PP
\&\fB\-v\fR
\&\fB\-\-version\fR
  Prints the version number and exits.
.SH "EXAMPLES"
.IX Header "EXAMPLES"
\&\fBmget \-cr url\fR
  Will download the video from the url, convert it and remove the leftover file without prompting the user on each step.
.PP
\&\fBmget \-dC url\fR
  Will download the movie, but will not attempt to convert it. Again the user will not be prompted.
.PP
\&\fBmget \-n cool_movie url\fR
  Will ask the user to download the movie or not. Will ask to convert the movie or not (if the movie is convertable). Will ask if it should remove the leftover files (if conversion occured). And will save the movie as cool_movie.extension where extension is determined automatically by the script.
.PP
\&\fBmget \-cr \-i file\fR
  The script will read movie links from the file line by line. It will download each movie, convert it if the movie is convertable and remove the leftover file. The downloaded files will be stored in sub\-directories.
.PP
\&\fBmget \-sD url\fR
  Will display a direct link that can be pasted into a browser and quit.
.SH "AUTHORS"
.IX Header "AUTHORS"
.Vb 2
\& Adam 'mulander' Wolk <netprobe@gmail.com>
\& defc0n <defc0n@da\-mail.net>
.Ve
.SH "COPYING"
.IX Header "COPYING"
.Vb 2
\& Copyright (C) 2006, 2007 Adam 'mulander' Wolk, defc0n.
\& Free use of this software is granted under the terms of the GNU General Public License (GPL).
.Ve
.SH "BUGS"
.IX Header "BUGS"
.Vb 2
\& If you want to report a bug or ask for a feature please do it on our google issue tracker
\& http://code.google.com/p/mget/issues/list
.Ve
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\&\fIwget\fR\|(1) \fIffmpeg\fR\|(1) \fImplayer\fR\|(1)


mget.rb
#!/usr/bin/ruby -w
# v1.50
#
# Copyright (C) 2006 Adam Wolk "mulander" <netprobe@gmail.com>
#                                "defc0n" <defc0n@da-mail.net>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA

$LOAD_PATH.push('lib')

require 'getoptlong'
require 'mget/mget'
require 'mget/config'

class Mget
  include Config
end

opts = GetoptLong.new(
  ["--help","-h",       GetoptLong::NO_ARGUMENT],
  ["--name","-n",       GetoptLong::REQUIRED_ARGUMENT],
  ["--input","-i",      GetoptLong::REQUIRED_ARGUMENT],
  ["--download","-d",   GetoptLong::NO_ARGUMENT],
  ["--nodownload","-D", GetoptLong::NO_ARGUMENT],
  ["--convert","-c",    GetoptLong::NO_ARGUMENT],
  ["--remove","-r",     GetoptLong::NO_ARGUMENT],
  ["--noremove","-R",   GetoptLong::NO_ARGUMENT],
  ["--noconvert","-C",  GetoptLong::NO_ARGUMENT],
  ["--show","-s",       GetoptLong::NO_ARGUMENT],
  ["--quiet","-q",      GetoptLong::NO_ARGUMENT],
  ["--version","-v",      GetoptLong::NO_ARGUMENT]
)

opts.ordering = GetoptLong::REQUIRE_ORDER

mget = Mget.new()
begin
  opts.each do |opt, arg|
    case opt
      when '--help'
        Mget.help()
      when '--name'
        mget.name   = arg
      when '--input'
        mget.input  = arg
      when '--download'
        mget.download = true
      when '--nodownload'
        mget.download = false
        mget.show     = true
      when '--convert'
        mget.convert  = true
      when '--noconvert'
        mget.convert  = false
      when '--remove'
        mget.remove   = true
      when '--noremove'
        mget.remove   = false
      when '--show'
        mget.show     = true
      when '--quiet'
        mget.quiet    = true
      when '--version'
        Mget.version()
    end
  end
rescue GetoptLong::InvalidOption
  Mget.help()
end
mget.target = ARGV[0] unless mget.fromFile?

mget.loadConfig()
mget.run()
mget.saveConfig()